packages/autotest/src/server/RouteHandler.ts

```diff
@@ -200,15 +202,42 @@ export default class RouteHandler {
             const remote = token ? body.remote.replace("https://", "https://" + token + "@") : body.remote;
             const tag = body.tag;
             const file = body.file;
-            const stream = await docker.buildImage(null, {remote, t: tag, dockerfile: file});
-            stream.on("error", (err: Error) => {
-                Log.error("Error building image. " + err.message);
-                res.send(500, "Error building image. " + err.message);
-            });
-            stream.on("end", () => {
-                Log.info("Finished building image.");
-            });
-            stream.pipe(res);
+
+            // Temporary: Work-around of 'dockerode' issue: https://github.com/apocas/dockerode/issues/536
+            // Uncomment when fixed:
+
+            // const stream = await docker.buildImage(null, {remote, t: tag, dockerfile: file});
+            // stream.on("error", (err: Error) => {
+            //     Log.error("Error building image. " + err.message);
+            //     res.send(500, "Error building image. " + err.message);
+            // });
+            // stream.on("end", () => {
+            //     Log.info("Finished building image.");
+            // });
+            // stream.pipe(res);
+
+            const dockerOptions = {remote, t: tag, dockerfile: file};
+            const reqParams = querystring.stringify(dockerOptions);
+            const reqOptions = {
+                socketPath: '/var/run/docker.sock',
+                path: '/v1.24/build?' + reqParams,
```

Comment:
A quick thought: we may want to double-check that our Docker versions support this API in production. This may mean that we want our Docker versions to be aligned in the near future (if not already).

Reply:
I can confirm that `210` and `310` are both at Docker version `19.03.2`, build `6a30dfc`.